packages/fxa-settings/src/components/PageDeleteAccount/index.test.tsx

```diff
@@ -80,19 +80,39 @@ describe('PageDeleteAccount', () => {
     expect(screen.getByTestId('continue-button')).toBeDisabled();
   });
 
-  it('Gets valid response on submit', async () => {
+  it('Enables "Delete" button once the password length is of 8 characters or more', async () => {
     renderWithRouter(
       <AppContext.Provider value={mockAppContext({ account })}>
         <PageDeleteAccount />
       </AppContext.Provider>
     );
 
     await advanceStep();
-    await typeByTestIdFn('delete-account-confirm-input-field')('hunter6');
+    expect(screen.getByTestId('delete-account-button')).toBeDisabled();
 
-    const deleteAccountButton = screen.getByTestId('delete-account-button');
-    expect(deleteAccountButton).toBeEnabled();
+    await act(async () => {
+      await typeByTestIdFn('delete-account-confirm-input-field')('hunter67');
+      expect(
+        screen.getByTestId('delete-account-confirm-input-field')['value'].length
+      ).toBeGreaterThanOrEqual(8);
+    });
+
+    expect(screen.getByTestId('delete-account-button')).toBeEnabled();
+  });
+
+  it('Does not enable "Delete" button if password length is less than 8 characters', async () => {
+    renderWithRouter(
+      <AppContext.Provider value={mockAppContext({ account })}>
+        <PageDeleteAccount />
+      </AppContext.Provider>
+    );
+
+    await advanceStep();
+
+    await act(async () => {
+      await typeByTestIdFn('delete-account-confirm-input-field')('hunter6');
+    });
 
-    expect(location.pathname).toContainEqual('/');
+    expect(screen.getByTestId('delete-account-button')).toBeDisabled();
```

Comment:
Heh, sorry if I over-complicated this. I think your tests could basically look like this:

```diff
diff --git a/packages/fxa-settings/src/components/PageDeleteAccount/index.test.tsx b/packages/fxa-settings/src/components/PageDeleteAccount/index.test.tsx
index 6fb6f2c23..207206e77 100644
--- a/packages/fxa-settings/src/components/PageDeleteAccount/index.test.tsx
+++ b/packages/fxa-settings/src/components/PageDeleteAccount/index.test.tsx
@@ -80,6 +80,19 @@ describe('PageDeleteAccount', () => {
     expect(screen.getByTestId('continue-button')).toBeDisabled();
   });
 
+  it('Enables "Delete" button once the password length is of 8 characters or more', async () => {
+    renderWithRouter(
+      <AppContext.Provider value={mockAppContext({ account })}>
+        <PageDeleteAccount />
+      </AppContext.Provider>
+    );
+
+    await advanceStep();
+    expect(screen.getByTestId('delete-account-button')).toBeDisabled();
+    await typeByTestIdFn('delete-account-confirm-input-field')('password');
+    expect(screen.getByTestId('delete-account-button')).toBeEnabled();
+  });
+
   it('Gets valid response on submit', async () => {
     renderWithRouter(
       <AppContext.Provider value={mockAppContext({ account })}>
@@ -88,7 +101,7 @@ describe('PageDeleteAccount', () => {
     );
 
     await advanceStep();
-    await typeByTestIdFn('delete-account-confirm-input-field')('hunter6');
+    await typeByTestIdFn('delete-account-confirm-input-field')('hunter67');
 
     const deleteAccountButton = screen.getByTestId('delete-account-button');
     expect(deleteAccountButton).toBeEnabled();

```

- Only updates the "Gets valid response on submit" test by adding one additional character so it's 8 characters long
- Removes the `act` wrapper in the new test, this is unnecesary
- Removes the assertion that the text field has 8 characters; we are supplying the text and know how many characters it will have, so we don't need to check that it contains the correct length
- We just want to check that it's first disabled, add the correct length of characters and check again that it's enabled

Reply:
@jodyheavener Oh I see. I thought that we need to test if the field is 8 characters long. Thanks for guiding me through it. I'll make the necessary changes and push it.